### ISSUE TITLE
Counting of tables starts from 2

### ISSUE THREAD

**Issue by souvikb1812** [bitslab/uic-cs-thesis-template#3]
Hello,  the counting of tables both in the list of tables and otherwise in the document starts from II instead of I. Is there any way to fix it ?

**Comment by jlazovskis** [comment]
Could you include some minimal code to explain what you mean? Perhaps this is an issue with additional packages you are using, because I am unable to reproduce this behavior.

Typesetting the code exactly as it is on this repo (no changes) produces one table (on raw page 12, formatted page 2) and one reference to it in the list of tables (raw page 7, formatted page vii). Both the caption of the table and the reference in the list of tables have "TABLE I".

**Comment by souvikb1812** [comment]
<img width="1280" alt="template_1" src="https://user-images.githubusercontent.com/14861348/219537384-7e4f0d2e-badf-428a-87c3-1900ad20744f.PNG">
<img width="1280" alt="template_2" src="https://user-images.githubusercontent.com/14861348/219537386-3f23e3b3-6bae-48a5-aba3-281c07585b18.PNG">

Hey, thanks for getting back to me and apologies for the delay. I am running the code as is without any alteration on Overleaf and I still see that the numbering starts from 2. I have attached two screenshots. Maybe it has something to do with Overleaf compatibility.

**Comment by pes10k** [comment]
Possibly? Im sorry i can't say more.  It's been 4 years since ive touched that tbh, and i never used it on overleaf at all.  If you find a fix that makes it work on overleaf AND work with a local build though, im happy to pull it into the repo so that others can have the fix as well!

**Comment by rushitnshah** [comment]
Hello!

Ran into a similar table numbering issue where the first table in the document appeared without a number. Numbering started incorrectly from second table. 

## TL;DR:
- Workaround: manually reset table counter right before the first table `\setcounter{table}{0}`.
- **Works with Overleaf (pdfLaTeX + TeXLive 2024).**
- This is only a (hack-ey) workaround.

## Issue:
The first table in my document appeared everywhere (including List of Figures) as:
> "TABLE : Caption for table 1 goes here"

instead of:
> "TABLE 1: Caption for table 1 goes here"

And, what was actually Table 2 was getting labelled Table 1 everywhere (including List of Figures) i.e.,
> "TABLE 1: Caption for table 2 goes here"

## Troubleshooting/Fix: 
Switching from `\documentclass{uicthesi}` to `\documentclass{article}` made Table 1 appear as Table 0. Assuming the numbering was off, I tried to manually set the table to counter just before the table using `\setcounter{table}{0}` and this resolved the issue. This is obviously not a great solution, but it works for now 🙃 My table 1 looks like this now

```tex
\setcounter{table}{0}   % manually set counter to 0
\begin{table}
    \begin{tabular}{ccc}
          ..........
    \end{tabular}
    \caption{.....}
    \label{.....}
\end{table}
```


I am no LaTeX expert but I thought this workaround might help narrow down the source of the issue.

**Comment by pes10k** [comment]
Hello @rushitnshah , thanks for the comment. Two thoughs:

1. if the issue actually is in `uicthesi`, it might be worth looking around and seeing if UIC has published an updated version of this somewhere in the last ~10 years. If so, I'll update it in this repo (if i still have write access… its been a long time)
2. if there isn't an updated `uicthesi` somewhere that fixes the issue, then can you:
    - see if the fix works if you do `\setcounter{table}{0}` anywhere in the document (instead of right before the first table)? If so, i'll just add the hack to the preamble (with a comment, and mentioning you and your fix if you wouldn't mind the credit)
    - it the fix needs to be right before the first `\begin{table}` then can you make a PR with your fix to `tables/ramones_stats.tex`, so at least things are documented in the files them selves (and to hopefully save future folks a headache)

**Comment by rushitnshah** [comment]
Thanks for your suggestions @pes10k. I think I found the issue!

Line 33 in `thesis.tex` manually sets the table counter to -1 right before building the table of contents:
```tex
% No need to make any changes to the next 10 lines
...
...
\setcounter{table}{-1}
\tableofcontents
...
...
```

Instead, using `\setcounter{table}{0}` fixes the table numbering. 
```tex
% No need to make any changes to the next 10 lines
...
...
\setcounter{table}{0}
\tableofcontents
...
...
```

If this seems like a reasonable solution for now, I'd like to submit a PR for this :) I can add an additional table in the template (like in the image below) so that this issue is apparent if it crops up again in the future.

![before_after_fix](https://github.com/user-attachments/assets/0c8d7f16-178e-422e-a271-4347e1fc77f8)


(Also, while the [graduate college thesis guidelines](https://uofi.app.box.com/s/ecf9vcm5g50kde0o4320cxevu9dwpagq) were last updated in 2019, I unfortunately could not find an accompanying LaTeX template. The [math, stats, cs department website](https://mscs.uic.edu/graduate/current-students/phd-thesis/) links to this repo directly.)

**Comment by pes10k** [comment]
ah, great find, that sounds great! If you could make a PR (and add a brief comment saying why the oddity is needed, and a link to this issue) i'd love to merge it.

Thanks for the find!
